packages/lit-dev-tools-esm/src/update-fonts.ts

```diff
@@ -0,0 +1,192 @@
+  console.log();
+  console.log(`Downloading ${font.name} from Google Fonts`);
+  // Fetch the font's CSS file from Google Fonts.
+  const gfCssUrl = `https://fonts.googleapis.com/css2?family=${
+    font.name
+  }:wght@${font.weights.join(';')}&display=swap`;
+  const gfCssResp = await fetch(gfCssUrl, {
+    headers: {
+      // We need to use a modern browser's user agent like Chrome, or else
+      // Google Fonts won't know that it can serve us a modern font format like
+      // woff2. We only support modern browsers on lit.dev, so there's no
+      // advantage to legacy fallback.
+      'User-Agent': CHROME_USER_AGENT,
+    },
+  });
+  if (!gfCssResp.ok) {
+    throw new Error(`Error fetching ${gfCssUrl}: ${gfCssResp.status}`);
+  }
+  const gfCssText = await gfCssResp.text();
+
+  // Process each @font-face rule in the Google Fonts CSS file.
+  const lowerKebabFontName = font.name.toLowerCase().replace(' ', '-');
+  const gfFaceRules = gfCssText.matchAll(/@font-face {.+?}/gs);
+  const litdevFaceRules = [];
+  const fontsToFetch = new Map<string, string>();
+  const weightsSet = new Set(font.weights);
+  for (const [gfRule] of gfFaceRules) {
+    if (font.latinOnly && !gfRule.includes(LATIN_UNICODE_RANGE)) {
+      continue;
+    }
+
+    // There will be a @font-face rule for each font-weight. We only care about
+    // some of them.
+    const weightMatch = gfRule.match(/font-weight: (\d+);/);
+    if (weightMatch === null) {
+      throw new Error(
+        `Could not extract weight from @font-face: ${gfRule} from ${gfCssUrl}`
+      );
+    }
+    const weight = Number(weightMatch[1]);
+    if (!weightsSet.has(weight)) {
+      continue;
+    }
+
+    // Extract the fonts.googleapis.com URL. We'll fetch this now, and then
+    // replace the url() with a path to our local copy.
+    const gfFontUrlMatch = gfRule.match(/src: url\((.+?)\)/);
+    if (gfFontUrlMatch === null) {
+      throw new Error(
+        `Could not extract src from @font-face: ${gfRule} from ${gfCssUrl}`
+      );
+    }
+    const gfFontUrl = gfFontUrlMatch[1];
+    const litdevFontPath = pathlib.join(
+      FONTS_DIR,
+      `${lowerKebabFontName}-${pathlib.basename(gfFontUrl)}`
+    );
+    fontsToFetch.set(gfFontUrl, litdevFontPath);
+    const litdevFontUrlPath =
+      '/' + pathlib.relative(LIT_DEV_SITE_DIR, litdevFontPath);
+    let litdevRule = gfRule.replace(gfFontUrl, litdevFontUrlPath);
+    if (font.latinOnly) {
+      // We don't need the unicode-range property in our @font-face rule because
+      // we don't have any non-latin characters to distinguish it from.
+      litdevRule = litdevRule.replace(LATIN_UNICODE_RANGE, '');
+    }
+    litdevFaceRules.push(litdevRule);
+  }
+
+  if (litdevFaceRules.length == 0) {
+    throw new Error(`Did not find any faces for font ${font.name}`);
+  }
+
+  // Fetch and write the font file.
+  for (const [gfFontUrl, litdevFontPath] of fontsToFetch) {
+    const fontResp = await fetch(gfFontUrl);
```

Comment:
can you put the loop body in an async function so the fonts can download in parallel?